src/includes/Services/CollectCmsDataService.php

```diff
@@ -0,0 +1,254 @@
+	 */
+	public function handle_collect_cms_data() {
+		if ( ! array_key_exists( 'HTTP_X_ALMA_SIGNATURE', $_SERVER ) ) {
+			$this->alma_logger->error( 'Header key X-Alma-Signature doesn\'t exist' );
+			$this->functions_proxy->send_http_response( array( 'error' => 'Header key X-Alma-Signature doesn\'t exist' ), 403 );
+			return;
+		}
+
+		try {
+			$this->security_helper->validate_collect_data_signature(
+				$this->alma_settings->get_active_merchant_id(),
+				$this->alma_settings->get_active_api_key(),
+				$_SERVER['HTTP_X_ALMA_SIGNATURE']
+			);
+			$this->functions_proxy->send_http_response( $this->payload_formatter->formatConfigurationPayload( $this->get_cms_info(), $this->get_cms_features() ), 200 );
+		} catch ( AlmaInvalidSignatureException $e ) {
+			$this->alma_logger->error( $e->getMessage() );
+			$this->functions_proxy->send_http_response( array( 'error' => $e->getMessage() ), 403 );
+		}
+	}
+
+	/**
+	 * Get CMS features.
+	 *
+	 * @return CmsFeatures
+	 */
+	private function get_cms_features() {
+		return new CmsFeatures(
+			array(
+				'alma_enabled'             => $this->alma_settings->is_enabled(),
+				'widget_cart_activated'    => 'yes' === $this->alma_settings->display_cart_eligibility,
+				'widget_product_activated' => 'yes' === $this->alma_settings->display_product_eligibility,
+				'used_fee_plans'           => $this->format_fee_plans(),
+				'in_page_activated'        => $this->alma_settings->display_in_page,
+				'log_activated'            => 'yes' === $this->alma_settings->debug,
+				'excluded_categories'      => $this->alma_settings->excluded_products_list,
+				'is_multisite'             => is_multisite(),
+			)
+		);
+	}
+
+	/**
+	 * Get CMS info.
+	 *
+	 * @return CmsInfo
+	 */
+	private function get_cms_info() {
+		return new CmsInfo(
+			array(
+				'cms_name'              => 'WooCommerce',
+				'cms_version'           => $this->option_proxy->get_option( 'woocommerce_version' ),
+				'third_parties_plugins' => $this->format_third_party_modules(),
+				'theme_name'            => $this->theme_proxy->get_name(),
+				'theme_version'         => $this->theme_proxy->get_version(),
+				'language_name'         => 'PHP',
+				'language_version'      => phpversion(),
+				'alma_plugin_version'   => ALMA_VERSION,
+				'alma_sdk_version'      => Client::VERSION,
+				'alma_sdk_name'         => 'alma/alma-php-client',
+			)
+		);
+	}
+
+	/**
+	 * Format third party modules.
+	 *
+	 * @return array
+	 */
+	private function format_third_party_modules() {
+		$active_plugins      = $this->option_proxy->get_option( 'active_plugins' );
+		$third_party_modules = array();
+		foreach ( $active_plugins as $plugin_data ) {
+			$third_party_modules[] = array( 'name' => $plugin_data );
+		}
+
+		return $third_party_modules;
+	}
+
+	/**
+	 * Format fee plans.
+	 *
+	 * @return array
+	 */
+	private function format_fee_plans() {
+		$plans = array();
+
+		if ( empty( $this->alma_settings->allowed_fee_plans ) ) {
+			return $plans;
```

Comment:
We concluded with @Francois-Gomis to return null if array plan is empty

Reply:
Done